relay-server/src/endpoints/common.rs

```diff
@@ -518,14 +519,16 @@ where
         .or_else(move |error: BadStoreRequest| {
             metric!(counter(RelayCounters::EnvelopeRejected) += 1);
 
-            if is_event {
+            let event_category = *event_category.borrow();
+            if event_category.is_some() {
```

Comment:
Requesting a Rust style check... is this the prettiest way to write this?

I considered
```rust
if let Some(event_category) = *event_category.borrow() {
```
but then line 531 needs to change to
```rust
event_category: Some(event_category),
```
which suggests that we don't want to unwrap the `Option` just to wrap it again.

Also, reusing the name `event_category` in a new `let` statement seems dubious by other languages' standards, but Rust seems to be cool with it and there's no better name for the value.

Reply:
> Also, reusing the name `event_category` in a new `let` statement seems dubious by other languages' standards, but Rust seems to be cool with it and there's no better name for the value.

Yeah. Thanks to strict typing and scoping rules we make use of this pattern in exactly such cases. Really all you do here is unbox a value, so I think it's totally fair 👍 

> I considered [...] but then line 531 needs to change to [...] `Some(event_category)` [...] which suggests that we don't want to unwrap the Option just to wrap it again.

That's precisely what you're doing, though. You're statically ensuring that you never pass `None` in. Using `Some(event_category)` below makes this even more explicit. It's not a bad thing to make it stick out that we really want to pass a `DataCategory` here. Both are equally fine to me, though, so your choice.

This gives me another idea. While we want to keep it an option on the raw outcome struct, have you considered making it non-optional on `TrackOutcome`? I believe at no place in the code we should be emitting `data_category: None` anymore if we're doing it right. This would be a neat way to enforce this.